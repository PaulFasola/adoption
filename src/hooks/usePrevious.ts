import { useRef, useEffect } from 'react';
import type { MutableRefObject } from 'react';

export function usePrevious<T>(state: T): MutableRefObject<T | undefined>['current'] {
  const ref = useRef<T>();

  useEffect(() => {
    ref.current = state;
  }, [state]);

  return ref.current;
}
