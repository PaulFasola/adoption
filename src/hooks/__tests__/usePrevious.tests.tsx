import React, { useState } from 'react';
import { configure, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';
import { usePrevious } from '../usePrevious';
import { toHaveNoViolations } from 'jest-axe';

import 'jest-styled-components';

expect.extend(toHaveNoViolations);
configure({ adapter: new Adapter() });

const TestComponent: React.FC = () => {
  const [counter, setCounter] = useState<number>(0);
  const previousCounter = usePrevious<number>(counter);

  return (
    <div>
      Previous State: {previousCounter}
      Current State: {counter}
      <button onClick={() => setCounter(counter + 1)}>Increment</button>
    </div>
  );
};
it('usePrevious hooks should work', () => {
  const wrapper = mount(<TestComponent />);
  expect(toJson(wrapper)).toMatchSnapshot();

  wrapper.find('button').simulate('click');
  wrapper.find('button').simulate('click');

  expect(toJson(wrapper)).toMatchSnapshot();
});
