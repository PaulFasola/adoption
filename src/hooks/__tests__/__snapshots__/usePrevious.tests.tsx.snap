// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usePrevious hooks should work 1`] = `
<TestComponent>
  <div>
    Previous State: 
    Current State: 
    0
    <button
      onClick={[Function]}
    >
      Increment
    </button>
  </div>
</TestComponent>
`;

exports[`usePrevious hooks should work 2`] = `
<TestComponent>
  <div>
    Previous State: 
    1
    Current State: 
    2
    <button
      onClick={[Function]}
    >
      Increment
    </button>
  </div>
</TestComponent>
`;
